Framework/Geometry/inc/MantidGeometry/Instrument/DetectorGroup.h

```diff
@@ -229,7 +229,7 @@ class MANTID_GEOMETRY_DLL DetectorGroup : public virtual IDetector {
   void initDraw() const override {}
 
   /// Returns the shape of the Object
-  const std::shared_ptr<const IObject> shape() const override {
+  const std::shared_ptr<const IObject> &shape() const override {
```

Comment:
Similar comment to `IObjComponent.h`. We can see that `DetectorGroup` is returning a new pointer, which is incompatible with a reference.

There are similar comments on anything inheriting `IObjComponent.h` below so I won't duplicate them later.